OpenOversight/tests/conftest.py

```diff
@@ -314,17 +314,16 @@ def db(app):
 @pytest.fixture(scope="function")
 def session(db):
     """Creates a new database session for a test."""
-    connection = db.engine.connect()
-    transaction = connection.begin()
-
-    session = scoped_session(session_factory=sessionmaker(bind=connection))
-    db.session = session
-
-    yield session
-
-    transaction.rollback()
-    connection.close()
-    session.remove()
+    with db.engine.connect() as connection:
+        with connection.begin():
+            session = scoped_session(session_factory=sessionmaker(bind=connection))
+            db.session = session
+
+            try:
+                yield session
+            finally:
+                session.remove()
+                connection.close()
```

Comment:
Sorry, quick comment - we shouldn't need this `connection.close` anymore since the context manager should automatically close the connection for us

Reply:
I remember trying to remove both of those individually when I made the change and I got some failed tests. If i remove the `connection.close()` statement, I get 53 failed tests.

```console
============================================================================ 53 failed, 436 passed, 2 skipped, 12 errors in 21.03s =============================================================================
```